File: Java/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int S=0;
        int l=0;
        int r = height.length-1;
        while(l<r){
            int h=Math.min(height[l],height[r]);
            S=Math.max(S,h*(r-l)); 
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return S;
    }
}

File: C#/11. Container With Most Water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int S=0;
        int l=0;
        int r = height.Length-1;
        while(l<r){
            int h=Math.Min(height[l],height[r]);
            S=Math.Max(S,h*(r-l)); 
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return S;
    }
}
